nodejs/server.js

```diff
@@ -0,0 +1,254 @@
+const express = require('express');
+const cors = require('cors');
+const bodyParser = require('body-parser');
+const app = express();
+const crate = require('node-crate');
+const uniqid = require('uniqid');
+const sha1 = require('sha1');
+
+const request = require('request');
+
+app.use(cors());
+app.use(bodyParser.json({limit: '50mb'}));
+app.use(bodyParser.urlencoded({limit: '50mb', extended: true, parameterLimit: 1000000}));
+
+crate.connect('localhost', 4200);
+
+app.listen(8080, () => console.log('Example app listening on port 8080!'));
+
+app.get('/', (req, response) => response.json({
+    message: 'Welcome to the node.js example for crate db',
+    state: 'Server up and running'
+}));
+
+// get all posts
+app.get('/posts', (req, response) => {
+    console.log(req.method + ' ' + req.url);
+    crate.execute("SELECT p.*, c.name as country, c.geometry as area FROM guestbook.posts AS p, guestbook.countries AS c WHERE within(p.\"user\"['location'], c.geometry) ORDER BY p.created DESC").then((res) => {
+        response.status(200).json(res.json);
+    })
+});
+
+// get a post by id
+app.get('/post/:id', (req, response) => {
+    console.log(req.method + ' ' + req.url);
+    crate.execute("SELECT p.*, c.name as country, c.geometry as area FROM guestbook.posts AS p, guestbook.countries AS c WHERE within(p.\"user\"['location'], c.geometry) AND p.id = ?", [req.params.id]).then((res) => {
+        response.status(200).json(res.json[0]);
+    })
+});
+
+// insert new post
+app.post('/posts', (req, response) => {
+    console.log(req.method + ' ' + req.url);
+    if (!req.body) {
+        response.json({
```

Comment:
is the rest of the function still executed? if yes, is that necessary? other than that add `return response.json({`

Reply:
and please check that for the whole file